pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -476,3 +482,59 @@ func (v *VPCConnectivity) getConnection(c *VPCConfig, src, dst Node) (conn *deta
 	}
 	return conn, nil
 }
+
+func (details *rulesAndConnDetails) updateRespondRules(c *VPCConfig, connQuery *connection.Set) error {
+	for _, srcDstDetails := range *details {
+		// respond rules are relevant if connection has a TCP component and non-stateful filter (NACL at the moment)
+		// are relevant for <src, dst>
+		if !respondRulesRelevant(srcDstDetails.conn, srcDstDetails.filtersRelevant) {
+			continue
+		}
+		connForResp := newTCPSet()
```

Comment:
calculation of connForResp can get out of loop, right?

Reply:
Not following. Lets take this f2f